File: ui/src/types/activity.ts
export interface ActivityEvent {
  id: number;
  event_type: string;
  timestamp: string;
  payload: Record<string, unknown>;
  task_title: string | null;
  step_title: string | null;
}

export interface ActivityResponse {
  run_id: string;
  events: ActivityEvent[];
  has_more: boolean;
}

export interface ClarificationRequestedPayload {
  event_type: 'clarification_requested';
  run_id: string;
  task_id: string;
  request_id: string;
  question_count: number;
}

export interface ClarificationRespondedPayload {
  event_type: 'clarification_responded';
  run_id: string;
  task_id: string;
  request_id: string;
}

File: src/orchestrator/api/schemas/activity.py
"""Activity feed API schemas."""

from datetime import datetime
from typing import Any

from orchestrator.api.schemas.base import ApiModel


class ActivityEvent(ApiModel):
    id: int
    event_type: str
    timestamp: datetime
    payload: dict[str, Any]
    task_title: str | None = None
    step_title: str | None = None


class ActivityResponse(ApiModel):
    run_id: str
    events: list[ActivityEvent]
    has_more: bool
